tests/v2/validation/provisioning/rke2/custom_cluster_test.go

```diff
@@ -112,8 +117,11 @@ func (c *CustomClusterProvisioningTestSuite) ProvisioningRKE2CustomCluster(exter
 					client, err := tt.client.WithSession(testSession)
 					require.NoError(c.T(), err)
 
-					numNodes := len(tt.nodeRoles)
-					nodes, err := externalNodeProvider.NodeCreationFunc(client, numNodes)
+					numNodesLin := len(tt.nodeRoles)
+
+					linuxNodes, err := externalNodeProvider.NodeCreationFunc(client, numNodesLin)
+					require.NoError(c.T(), err)
+					winNodes, err := externalNodeProvider.WinNodeCreationFunc(client, tt.nodeCountWin, tt.hasWindows)
```

Comment:
Yeah I agree, I think those if statements you have in the extension belong here at the test level. Having those if statements in the actual extensions is not a correct place in my opinion.

Reply:
I have restructured the code so that this if statement is gone, and the check for isWindows makes more sense now. I was also able to strip out a TON of no longer needed code as a result of my refactor.